### PR TITLE
Added unixbench benchmarking test

### PR DESCRIPTION
**Pull request by santhosh93**
Added unix bench benchmarking test to the framework.
Added a new Makefile.patch which make the compilation of unixbench successfull.
The test runs the benchmark and checks for the error if occurred in between and finally displays the benchmarking score.

### CODE REVIEW

File: perf/unixbench.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            self.error('Gcc is needed for the test to be run')
+        if not sm.check_installed("make") and not sm.install("make"):
+            self.error('make is needed for the test to be run')
+        if detected_distro.name == "SuSE":
+            if not sm.check_installed("git-core") and not sm.install("git-"
+                                                                     "core"):
+                self.error('git is needed for the test to be run')
+        else:
+            if not sm.check_installed("git") and not sm.install("git"):
+                self.error('git is needed for the test to be run')
```
Comment (PraveenPenguin) [inline perf/unixbench.py]: Too much repetition on the same check. Could you simplify this? Something like this:
https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/generic/gdb.py#L33-L43

File: perf/unixbench.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        git.get_repo('https://github.com/kdlucas/byte-unixbench.git',
+                     destination_dir=self.srcdir)
+        os.chdir(self.srcdir + '/UnixBench')
```
Comment (PraveenPenguin) [inline perf/unixbench.py]: can this can be 
self.srcdir=os.path.join(self.srcdir,'UnixBench'))
os.chdir(self.srcdir)

File: perf/unixbench.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+from avocado.utils import distro
+from avocado.core import data_dir
+
+
+class unixbench(Test):
+    def setUp(self):
+        # Check for root permission
+        if os.geteuid() != 0:
+            exit("You need to have root privileges to run this script."
+                 "\nPlease try again, using 'sudo'. Exiting.")
```
Comment (PraveenPenguin) [inline perf/unixbench.py]: can `sudo=True` will be used ?

**Comment by santhosh93** [comment]
Opened a new pull Request : https://github.com/avocado-framework-tests/avocado-misc-tests/pull/34

**Pull request closed without merging**